src/org/scratchOrgInfoGenerator.ts

```diff
@@ -216,12 +216,16 @@ export const generateScratchOrgInfo = async ({
     // project is not required
   }
 
+  const { namespace: originalNamespace, ...payload } = scratchOrgInfoPayload;
+
+  const namespace = originalNamespace ?? sfProject?.get('namespace');
```

Comment:
so, `originalNamespace` comes from the definition file and takes precedence over `namespace` defined in sfdx-project.json. 

it's doc'd for sfdx-project json but not for the definition file here
https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_scratch_orgs_def_file.htm

I'll ask Juliet if we should add a note about it in docs.